source/plugins/go/src/telemetry.go

```diff
@@ -556,17 +556,27 @@ func PushToAppInsightsTraces(records []map[interface{}]interface{}, severityLeve
 		} else if strings.Contains(logEntry, "E! [inputs.prometheus]") {
 			populateKubeMonAgentEventHash(record, PromScrapingError)
 		} else if strings.Contains(logEntry, "Lifetime validation failed. The token is expired.") {
-			UpdateMdsdErrorMetrics("MdsdTokenExpired")
+			UpdateTracesErrorMetrics("MdsdTokenExpired")
 		} else if strings.Contains(logEntry, "Failed to upload to ODS: Error resolving address") {
-			UpdateMdsdErrorMetrics("MdsdODSUploadErrorResolvingAddress")
+			UpdateTracesErrorMetrics("MdsdODSUploadErrorResolvingAddress")
 		} else if strings.Contains(logEntry, "Data collection endpoint must be used to access configuration over private link") {
-			UpdateMdsdErrorMetrics("MdsdPrivateLinkNoDCE")
+			UpdateTracesErrorMetrics("MdsdPrivateLinkNoDCE")
 		} else if strings.Contains(logEntry, "Failed to register certificate with OMS Homing Service:Error resolving address") {
-			UpdateMdsdErrorMetrics("MdsdOMSHomingServiceError")
+			UpdateTracesErrorMetrics("MdsdOMSHomingServiceError")
 		} else if strings.Contains(logEntry, "Could not obtain configuration from") {
-			UpdateMdsdErrorMetrics("MdsdGetConfigError")
+			UpdateTracesErrorMetrics("MdsdGetConfigError")
 		} else if strings.Contains(logEntry, " Failed to upload to ODS: 403") {
-			UpdateMdsdErrorMetrics("MdsdODSUploadError403")
+			UpdateTracesErrorMetrics("MdsdODSUploadError403")
+		} else if strings.Contains(logEntry, "failed getting access token") {
+			UpdateTracesErrorMetrics("AddonTokenAdapterFailedGettingAccessToken")
+		} else if strings.Contains(logEntry, "failed to watch token secret") {
+			UpdateTracesErrorMetrics("AddonTokenAdapterFailedToWatchTokenSecret")
+		} else if strings.Contains(logEntry, "http: Server closed") {
+			UpdateTracesErrorMetrics("AddonTokenAdapterServerClosed")
+		} else if strings.Contains(logEntry, "forwarding the token request to IMDS...") {
```

Comment:
questions (1) cant we drop all the logs everything except exceptions or failures or errors and ingest as traces? (2) doing log to metrics, how much telemetry logs will be dropped and are we seeing significant telemetry optimization with this change? (2) are we covering all the possible errors/failures/exceptions? (4) can you please check last 10 days telemetry to  check if we are missing anything? (5) can we separate metric for private link scenario?

Reply:
1 - We can look for these keywords but what if some other keywords are used? 
2/3/4 - These are the major bucket of traces that I found so included them here. It will definitely help reduce the telemetry. I can get some specific numbers as a graph if needed. 
5 - what do you mean by private link scenario? As in the number of these errors where the cluster is private link or are you looking for some specific trace?

Reply:
1.  Please take look at the addon token adapter code and reviewing the code covering "fail" and "error" should cover all the possible scenarios 2. I think, we can completely turnoff all the verbose logs and just capture either metric or logs for failed scenarios.